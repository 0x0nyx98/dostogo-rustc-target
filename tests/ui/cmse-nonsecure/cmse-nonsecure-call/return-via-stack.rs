//@ build-fail
//@ compile-flags: --target thumbv8m.main-none-eabi --crate-type lib
//@ needs-llvm-components: arm
#![feature(abi_c_cmse_nonsecure_call, no_core, lang_items, intrinsics)]
#![no_core]
#[lang = "sized"]
pub trait Sized {}
#[lang = "copy"]
pub trait Copy {}
impl Copy for u32 {}

#[repr(C)]
pub struct ReprCU64(u64);

#[repr(C)]
pub struct ReprCBytes(u8, u8, u8, u8, u8);

#[repr(C)]
pub struct U64Compound(u32, u32);

#[repr(C, align(16))]
pub struct ReprCAlign16(u16);

#[no_mangle]
pub fn test(
    f1: extern "C-cmse-nonsecure-call" fn() -> ReprCU64,
    f2: extern "C-cmse-nonsecure-call" fn() -> ReprCBytes,
    f3: extern "C-cmse-nonsecure-call" fn() -> U64Compound,
    f4: extern "C-cmse-nonsecure-call" fn() -> ReprCAlign16,
    f5: extern "C-cmse-nonsecure-call" fn() -> [u8; 5],
    f6: extern "C-cmse-nonsecure-call" fn() -> u128, //~ WARNING [improper_ctypes_definitions]
    f7: extern "C-cmse-nonsecure-call" fn() -> i128, //~ WARNING [improper_ctypes_definitions]
) {
    f1(); //~ ERROR [E0798]
    f2(); //~ ERROR [E0798]
    f3(); //~ ERROR [E0798]
    f4(); //~ ERROR [E0798]
    f5(); //~ ERROR [E0798]
    f6(); //~ ERROR [E0798]
    f7(); //~ ERROR [E0798]
}
