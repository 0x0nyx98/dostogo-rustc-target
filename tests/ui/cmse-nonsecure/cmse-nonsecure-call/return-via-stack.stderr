warning: `extern` fn uses type `u128`, which is not FFI-safe
  --> $DIR/return-via-stack.rs:31:9
   |
LL |     f6: extern "C-cmse-nonsecure-call" fn() -> u128,
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not FFI-safe
   |
   = note: 128-bit integers don't currently have a known stable ABI
   = note: `#[warn(improper_ctypes_definitions)]` on by default

warning: `extern` fn uses type `i128`, which is not FFI-safe
  --> $DIR/return-via-stack.rs:32:9
   |
LL |     f7: extern "C-cmse-nonsecure-call" fn() -> i128,
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not FFI-safe
   |
   = note: 128-bit integers don't currently have a known stable ABI

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:34:5
   |
LL |     f1: extern "C-cmse-nonsecure-call" fn() -> ReprCU64,
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f1();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:35:5
   |
LL |     f2: extern "C-cmse-nonsecure-call" fn() -> ReprCBytes,
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f2();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:36:5
   |
LL |     f3: extern "C-cmse-nonsecure-call" fn() -> U64Compound,
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f3();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:37:5
   |
LL |     f4: extern "C-cmse-nonsecure-call" fn() -> ReprCAlign16,
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f4();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:38:5
   |
LL |     f5: extern "C-cmse-nonsecure-call" fn() -> [u8; 5],
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f5();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:39:5
   |
LL |     f6: extern "C-cmse-nonsecure-call" fn() -> u128,
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f6();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error[E0798]: return value of `C-cmse-nonsecure-call` function too large to pass via registers
  --> $DIR/return-via-stack.rs:40:5
   |
LL |     f7: extern "C-cmse-nonsecure-call" fn() -> i128,
   |     -- this function uses the `C-cmse-nonsecure-call` ABI
...
LL |     f7();
   |     ^^^^ but its return value doesn't fit in the available registers
   |
   = note: functions with the `C-cmse-nonsecure-call` ABI must pass their result via the available return registers

error: aborting due to 7 previous errors; 2 warnings emitted

For more information about this error, try `rustc --explain E0798`.
